datashader/transfer_functions.py

```diff
@@ -412,8 +412,9 @@ def spread(img, px=1, shape='circle', how='over', mask=None, name=None):
         Optional string name to give to the Image object to return, 
         to label results for display.
     """
-    if not isinstance(img, Image):
-        raise TypeError("Expected `Image`, got: `{0}`".format(type(img)))
+    if not isinstance(img, xr.DataArray):
+        raise TypeError("Expected `xr.DataArray`, got: `{0}`".format(type(img)))
+    is_image = isinstance(img, Image)
```

Comment:
Be sure to test with categorical arrays; those will have is_image be False but need a loop to iterate over the Z dimension, presumably spreading in each one separately?  Or maybe they should spread analogously to RGB, where instead of 3 layers R,G,B they have arbitrarily many layers, all spread at once?  Seems tricky.

Reply:
This is now resolved: categorical aggregates are supported by spreading each category layer by the same amount.